src/DotLiquid/Condition.cs

```diff
@@ -19,7 +19,7 @@ public class Condition
     {
         #region Condition operator delegates
 
-        public static readonly Dictionary<string, ConditionOperatorDelegate> Operators = new Dictionary<string, ConditionOperatorDelegate>(Template.NamingConvention.StringComparer)
+        public static readonly Dictionary<string, ConditionOperatorDelegate> Operators = new Dictionary<string, ConditionOperatorDelegate>(Template.NamingConvention.ConditionComparer)

```

Comment:
Since the comparer is used in the creation of a static field, it will not change depending on the selected naming convention.
I think it could be replaced by a custom `IEqualityComparer` based on the current Template.NamingConvention.